tools/downloader/src/open_model_zoo/model_tools/converter.py

```diff
@@ -104,11 +109,30 @@ def main():
 
     args = parser.parse_args()
 
+    try:
+        telemetry = tm.Telemetry('Model Converter', tid='UA-194864834-1')
+        telemetry.start_session('md')
+
+        if args.all:
+            telemetry.send_event('md', 'converter_selection_mode', 'all')
+        if args.list:
+            telemetry.send_event('md', 'converter_selection_mode', 'list')
+        if args.name:
+            telemetry.send_event('md', 'converter_selection_mode', 'name')
+    except NameError:
+        pass
+
     mo_path = args.mo
     if mo_path is None:
         try:
             mo_path = Path(os.environ['INTEL_OPENVINO_DIR']) / 'deployment_tools/model_optimizer/mo.py'
         except KeyError:
+            try:
+                telemetry.send_event('md', 'converter_error', str({'error': 'Unable to locate Model Optimizer.', 'model': 'none'}))
+                telemetry.end_session('md')
+                telemetry.force_shutdown(1.0)
```

Comment:
I see this fragment everywhere; perhaps it would be better to wrap all code covered by the telemetry in a `try` block and move this code to the `finally` block?

(You can factor out the contents of the `try` block into a new function so that you don't have to reindent everything.)

Reply:
Done